src/js/actions/transform.js

```diff
@@ -1136,6 +1136,40 @@ define(function (require, exports) {
     rotateLayersInCurrentDocument.transfers = [rotate];
 
     /**
+     * Handle the "editArtboardEvent" from photoshop
+     * TODO: This is a bit heavy-handed
+     *
+     * @return {Promise}
+     */
+    var handleTransformArtboard = function () {
+        // After each action we call, document model changes
+        // so we re-get it
+        var appStore = this.flux.store("application"),
+            nextDoc = appStore.getCurrentDocument();
+
+        if (!nextDoc) {
+            throw new Error("No current document");
+        }
+
+        return this.transfer("history.newHistoryStateRogueSafe", nextDoc.id)
+            .bind(this)
+            .then(function () {
+                return this.transfer("layers.resetBounds", nextDoc, nextDoc.layers.allSelected);
+            })
+            .then(function () {
+                nextDoc = appStore.getCurrentDocument();
+                return this.transfer("layers.resetSelection", nextDoc);
+            }).then(function () {
```

Comment:
I fixed the indenting of `.then()`.  I stared at this block for awhile too, and made a quick TODO comment in the function header because it seems heavy-handed (and in a weird order! as you said) but I was hoping not to address it as part of this PR which is otherwise safe-ish.